### ISSUE TITLE
Support distinction between source and resource directories in IDEA

### ISSUE THREAD

**Issue by trejkaz** [gradle/gradle#1738]
Given a fairly trivial project:

```
$ cat build.gradle
apply plugin: 'java'
apply plugin: 'idea'

repositories {
    jcenter()
}

$ find src
src/
src/main
src/main/java
src/main/resources
```

If I run

```
gradle idea
```

It creates an `.iml` file containing this:

          <sourceFolder url="file://$MODULE_DIR$/src/main/resources" isTestSource="false"/>

But the line should actually be this:

          <sourceFolder url="file://$MODULE_DIR$/src/main/resources" type="java-resource" />

I assume this also occurs for test resources.

The inconvenience is that IDEA thinks these resources directories are source directories and they show up, for instance, every time you want to move a class to another location.

**Comment by trejkaz** [comment]
Thought I'd also add that there is also "Generated Sources", so ANTLR Plugin + IDEA Plugin creates a similar situation where generates sources are not marked as such, but it isn't clear whether that is entirely the IDEA plugin's fault.

**Comment by leonard84** [comment]
Yes this is a problem, furthermore it is not possible to set additional `resourceDirs`/`testResourceDirs`.

**Comment by fanick1** [comment]
I've implemented support for resource folders to be marked as resources - they are  now generated in the .iml as 
` <sourceFolder url="file://$MODULE_DIR$/src/main/resources" type="java-resource" />`
also the idea.module can now be configured with additional `resourceDirs ` and `testResourceDirs`
see my PR https://github.com/gradle/gradle/pull/3724

**Comment by blindpirate** [comment]
Closed by #3724

**Comment by fanick1** [comment]
Nice.
